firebase-appdistribution/src/test/java/com/google/firebase/appdistribution/ApkUpdaterTest.java

```diff
@@ -170,6 +183,52 @@ public void updateApk_whenInstallSuccessful_setsResult() throws Exception {
     assertThat(updateTask.isSuccessful()).isTrue();
   }
 
+  @Test
+  public void updateApk_whenSuccessfullyUpdated_notificationsSetCorrectly()
+      throws FirebaseAppDistributionException, ExecutionException, InterruptedException,
+          IOException {
+    doReturn(new ByteArrayInputStream(TEST_FILE.getBytes()))
+        .when(mockHttpsUrlConnection)
+        .getInputStream();
+    doNothing().when(apkUpdater).validateJarFile(any(), anyLong(), anyBoolean(), anyLong());
+    when(mockApkInstaller.installApk(any(), any())).thenReturn(Tasks.forResult(null));
+
+    UpdateTask updateTask = apkUpdater.updateApk(TEST_RELEASE, true);
+    updateTask.addOnCompleteListener(testExecutor, onCompleteListener);
+    List<UpdateProgress> events = new ArrayList<>();
+    updateTask.addOnProgressListener(testExecutor, events::add);
+    onCompleteListener.await();
+
+    assertThat(events).hasSize(3);
+    assertThat(events.get(0).getUpdateStatus()).isEqualTo(UpdateStatus.PENDING);
+    assertThat(events.get(1).getUpdateStatus()).isEqualTo(UpdateStatus.DOWNLOADING);
+    assertThat(events.get(1).getApkBytesDownloaded()).isEqualTo(TEST_FILE.length());
+    assertThat(events.get(2).getUpdateStatus()).isEqualTo(UpdateStatus.DOWNLOADED);
+  }
+
+  @Test
+  public void updateApk_invalidJarFile_throwsException() throws IOException {
+    doReturn(new ByteArrayInputStream(TEST_FILE.getBytes()))
+        .when(mockHttpsUrlConnection)
+        .getInputStream();
+    when(mockApkInstaller.installApk(any(), any())).thenReturn(Tasks.forResult(null));
+    FileInputStream fileInputStream =
+        (FileInputStream) ApkUpdaterTest.class.getResourceAsStream("file does not exist");
+    doReturn(fileInputStream).when(mockContext).openFileInput(anyString());
```

Comment:
For validateJarFile it takes in a File. I am unable to mock that method anymore I decided to pass in a file. We get that file from calling openFileInput on the context. So I had to mock the context so that it would return the correct FileinputStream.

Reply:
I'm confused because I don't see us calling `openFileInput` anywhere. If it's used by something else under the hood (maybe the `JarFile` constructor?) then we should probably leave a comment here explaining that.

Reply:
O sorry for the confusion that is an unnecessary mock. Thanks for catching that!